src/seedu/addressbook/data/person/Address.java

```diff
@@ -38,22 +55,127 @@ public static boolean isValidAddress(String test) {
 
     @Override
     public String toString() {
-        return value;
+        String address ="";
+        address+=block+SPLIT_TOKEN;
+        address+=street+SPLIT_TOKEN;
+        address+=unit+SPLIT_TOKEN;
+        address+=postalCode;
+
+        return address;
     }
 
     @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+                && this.toString().equals(((Address) other).toString())); // state check
     }
 
     @Override
     public int hashCode() {
-        return value.hashCode();
+        return this.toString().hashCode();
     }
 
     public boolean isPrivate() {
         return isPrivate;
     }
+
+    class Block {
+        public final String block;
+
+        public Block(String block) {
+            this.block = block;
+        }
+
+        @Override
+        public String toString() {
+            return block;
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            return other == this
+                    || (other instanceof Block
+                    && this.block.equals(((Block) other).block));
+        }
+
+        @Override
+        public int hashCode() {
+            return block.hashCode();
+        }
+    }
+    class Street {
+        private String value;
+
+        Street(String street) {
+            this.value = street;
+        }
+
+        @Override
+        public String toString() {
+            return value;
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            return other == this
+                    || (other instanceof Block
```

Comment:
Hmmm when copying code snippets be careful to change everything! This is not a `Block`...Same issue applies to `Unit` and `Postal Code`